Experimental/src/org/sleuthkit/autopsy/experimental/textclassifier/TextClassifierFileIngestModule.java

```diff
@@ -61,129 +38,85 @@
-        this.categorizer = new DocumentCategorizerME(doccatModel);
-    }
-    
-    @Messages({"ObjectDetectionFileIngestModule.noClassifiersFound.subject=No classifiers found.",
-        "# {0} - classifierDir", "ObjectDetectionFileIngestModule.noClassifiersFound.message=No classifiers were found in {0}, object detection will not be executed."})
+    private TextClassifierUtils utils;
+
+    @Messages({"TextClassifierFileIngestModule.noClassifiersFound.subject=No classifiers found.",
+        "# {0} - classifierDir", "TextClassifierFileIngestModule.noClassifiersFound.message=No classifiers were found in {0}, text classifier will not be executed."})
     @Override
     public void startUp(IngestJobContext context) throws IngestModule.IngestModuleException {
+        utils = new TextClassifierUtils();
+
         try {
-            this.fileTypeDetector = new FileTypeDetector();
-        } catch (FileTypeDetectorInitException ex) {
-            throw new IngestModule.IngestModuleException("Exception while constructing FileTypeDector.", ex);
-        }
-        this.tokenizer = SimpleTokenizer.INSTANCE;
-        
-        jobId = context.getJobId();
-       
-        try {
-            loadModel();
+            categorizer = TextClassifierUtils.loadModel();
         } catch (IOException ex) {
             throw new IngestModule.IngestModuleException("Unable to load model for text classifier module.", ex);
         }
-                
+
         try {
-            blackboard = Case.getCurrentCaseThrows().getServices().getBlackboard();
+            blackboard = Case.getCurrentCaseThrows().getServices().getArtifactsBlackboard();
         } catch (NoCurrentCaseException ex) {
             throw new IngestModule.IngestModuleException("Exception while getting open case.", ex);
         }
     }
-    
+
     @Override
     public ProcessResult process(AbstractFile file) {
-        isSupported(file);
-        
-        if (file.getSize() > MAX_FILE_SIZE) {
+        if (!utils.isSupported(file, this)) {
+            return ProcessResult.OK;
+        }
+
+        if (file.getSize() > TextClassifierUtils.MAX_FILE_SIZE) {
             //prevent it from allocating gigabytes of memory for extremely large files
             logger.log(Level.INFO, "Encountered file " + file.getParentPath() + file.getName() + " with object id of "
-                    + file.getId() + " which exceeds max file size of " + MAX_FILE_SIZE + " bytes, with a size of " + file.getSize());
+                    + file.getId() + " which exceeds max file size of " + TextClassifierUtils.MAX_FILE_SIZE + " bytes, with a size of " + file.getSize());
             return IngestModule.ProcessResult.OK;
         }
-        
-        boolean isInteresting;
+
+        boolean isNotable;
         try {
-            isInteresting = classify(file);
+            isNotable = classify(file);
         } catch (TextExtractorFactory.NoTextExtractorFound ex) {
```

Comment:
We can probably do a multi-catch with this since we do the same thing each time.  If we log the exception, then we'll know what type it was.